esx_service/vmdk_ops.py

```diff
@@ -778,6 +778,7 @@ def disk_attach(vmdk_path, vm):
             msg=("Failed to add PVSCSI Controller: %s", ex.msg)
             return err(msg)
         # Find the controller just added
+        devices = vm.config.hardware.device
```

Comment:
The PCI slot number isn't the controller key provided when configuring the PVSCSI controller to the VM. The PCI slot number is determined by the VM configuration task itself.

This is very much needed as the PCI slot number at which the controller is presented is what the guest determines when building its PCI device tree. And that helps us match what ESX used and what the guest found to figure where VMDKs get presented in the guest.